userspace/libsinsp/container_engine/cri.cpp

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright (C) 2013-2018 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "container_engine/cri.h"
+
+#include <sys/stat.h>
+#include <grpc++/grpc++.h>
+#include "cri.pb.h"
+#include "cri.grpc.pb.h"
+
+#include "container_engine/docker.h"
+#include "container_engine/mesos.h"
+#include <cri.h>
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+using namespace libsinsp::cri;
+
+libsinsp::container_engine::cri::cri()
+{
+	if(s_cri || s_cri_unix_socket_path.empty()) {
+		return;
+	}
+
+	auto cri_path = scap_get_host_root() + s_cri_unix_socket_path;
+	struct stat s;
+	if(stat(cri_path.c_str(), &s) != 0 || (s.st_mode & S_IFMT) != S_IFSOCK) {
+		return;
+	}
+
+	s_cri = runtime::v1alpha2::RuntimeService::NewStub(
+		grpc::CreateChannel("unix://" + cri_path, grpc::InsecureChannelCredentials()));
+
+	runtime::v1alpha2::VersionRequest vreq;
+	runtime::v1alpha2::VersionResponse vresp;
+
+	vreq.set_version("v1alpha2");
+	grpc::ClientContext context;
+	auto deadline = std::chrono::system_clock::now() + std::chrono::milliseconds(s_cri_timeout);
+	context.set_deadline(deadline);
+	grpc::Status status = s_cri->Version(&context, vreq, &vresp);
+
+	if (!status.ok())
+	{
+		// we could disable CRI support here...
+		return;
+	}
+
+	s_cri_runtime_type = get_cri_runtime_type(vresp.runtime_name());
```

Comment:
it'd be nice to log the relevant info in vresp, including the runtime name, at INFO level.

Reply:
Fixed